doc/files/opensc.conf.5.xml.in

```diff
@@ -772,6 +772,53 @@ app <replaceable>application</replaceable> {
 			</variablelist>
 		</refsect2>
 
+		<refsect2 id="piv">
+			<title>Configuration Options for PIV Card</title>
+			<variablelist>
+				<varlistentry>
+					<term>
+						<option>piv_max_object_size = <replaceable>num</replaceable>;</option>
+					</term>
+					<listitem><para>
+						Max size of a PIV object. The default of 16384 bytes should
+						work for most cards. It might be overwritten by
+						<literal>PIV_MAX_OBJECT_SIZE</literal> environment variable.
+					</para></listitem>
+				</varlistentry>
+				<varlistentry>
+					<term><option>piv_use_sm = <replaceable>name</replaceable>;</option>
+					</term>
+					<listitem><para>
+						Use optional secure messaging as define by NIST SP 800-73-4
+					<replaceable>name</replaceable> is one of:
+					<itemizedlist>
+						<listitem><para><literal>always</literal>
+							- Use secure messaging for commands.
+						</para></listitem>
+						<listitem><para><literal>never</literal>
+							- Used for debugging.  Card may still require use of secure messaging.
```

Comment:
`piv_use_sm` was added originally for testing, but can provide extra security. APDUs will not show up in plain text  in OpenSC, pcscd, USB trace or any other location as encryption is between card driver and card. 

NIST originally restricted access over contactless to a subset of commands.  With 800-74-4 the Virtual Contact Interface (VCI) uses SM over contactless. SM is also usable over contact interface.

The current code default  is only using  SM  if available over contactless and with VCI. The default could be changed to always  use SM if card supports it. 

If `piv_use_sm` is not set, SM will only be used with VCI. 

If `piv_use_sm=never` treat the card more like a 800-73-3 card. (This could be removed.)

If `piv_use_sm=always` use SM in all cases.

If card does not support SM, ignore `piv_use_sm`.

These 800-73-4 cards are much faster the previous cards, and using  `piv_use_sm=always` does not appear to couse delays.

Reply:
I understand the technical reasons why and where to switch on SM. What I'd like to point out is that if the user chooses `piv_use_sm=never`, SM should *never* be used as opposed to *maybe* not as described in the help text.

Reply:
>  SM should never be used as opposed to maybe not as described in the help text.

If "never"  tells the driver to not use SM even if card supports it.  "Card may still require use of secure messaging."  could be reworded to:  "The card will continue to enforce any policies that require the SM." or  "The use of "never" does not override any policies enforced by a card."  

For example Yubico allows some commands over contactless that are not allowed by  the NIST standards. The current card-piv.c will try to do command over contactless reader which will fail if card does not allow it. By default i.e. `piv_us_sm` is not set,  the PR will check if card supports SM and will use SM over contactless.

Reply:
https://github.com/OpenSC/OpenSC/pull/2053/commits/fce5c2ace83fc0e4f1f50a88c5f58669f40881b4 should address this.  (will push spelling fix).